extensions/amp-viewer-assistance/0.1/amp-viewer-assistance.js

```diff
@@ -148,6 +156,32 @@ export class AmpViewerAssistance {
     });
   }
 
+
+  /**
+   * Checks the 'actionStatus' field of the updateActionState arguments against
+   * a whitelist.
+   * @private
+   * @param {!Object} args
+   * @return {boolean}
+   */
+  isValidActionStatusArgs_(args) {
+    const update = args['update'];
+    if (!update || !update['actionStatus']) {
```

Comment:
Hm, not sure we support object literals outside of AMP.setState ([syntax](https://github.com/ampproject/amphtml/blob/master/spec/amp-actions-and-events.md#actions-and-events-in-amp)). What would this look like in HTML?

```html
<!-- Is this the intention? -->
<button on="tap:amp-viewer-assistance.updateActionStatus(update={actionStatus: ACTIVE_ACTION_STATUS})">

<!-- 
  Or maybe we can do this (case-insensitive)? 
  Uppercase and append "_ACTION_STATUS" before sending viewer message. 
-->
<button on="tap:amp-viewer-assistance.updateActionStatus(status=active)">
```

Reply:
The first example is how it's usually implemented except with the payload being passed from an event:

```
<form id="my-form"
      action-xhr="some/XHR"
      on="submit-success:amp-viewer-assistance.updateActionState(update=event.response.result)">
 </form>
```

It's also important to note that the `update` payload can include additional data in another field (usually `result`), but it is too complex to validate on the run-time side.
```
{
  'actionStatus': 'COMPLETED_ACTION_STATUS',
  'result': { ... }
}
```

Reply:
I see, thanks.